src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -294,8 +295,23 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	m_authProvider = new JenkinsAuthenticationProvider(m_credentials, build.getParent().getParent());
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
-    	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+        Map<String, String> properties = getScanProperties(build,listener);
+    	final IScan scan = ScanFactory.createScan(properties, progress, m_authProvider);
+        JenkinsAuthenticationProvider checkAppScan360 = (JenkinsAuthenticationProvider) m_authProvider;
+        if(checkAppScan360.isAppScan360()) {
+            if (m_type.equals("Dynamic Analyzer")) {
+                throw new AbortException(Messages.error_dynamic_analyzer_AppScan360());
+            } else if (m_intervention) {
+                progress.setStatus(new Message(Message.WARNING, Messages.warning_allow_intervention_AppScan360()));
+            } else if (properties.get("openSourceOnly") != null) {
+                throw new AbortException(Messages.error_OSO_AppScan360());
+            }
+        }
+
+        if(!checkAppScan360.isAppScan360() && m_authProvider.getacceptInvalidCerts()){
```

Comment:
Since this is the opposite check of the previous check, can you use an `else if`? For example:
`else if(m_authProvider.getacceptInvalidCerts()){`

Reply:
Okay. Now, doing all the 4 checks with the "if-else" statement.